flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -160,8 +170,37 @@ public static void setupYarnClassPath(Configuration conf, Map<String, String> ap
 
 		fs.copyFromLocalFile(false, true, localSrcPath, dst);
 
+		// Note: If we directly used registerLocalResource(FileSystem, Path) here, we would access the remote
+		//       file once again which has problems with eventually consistent read-after-write file
+		//       systems. Instead, we decide to wait until the remote file be available.
+
+		FileStatus[] fss = null;
+		int iter = 1;
+		while (iter <= REMOTE_RESOURCES_FETCH_NUM_RETRY) {
```

Comment:
Conceptually, what is a retry? If I set it to `0`, would you expect an initial attempt and then no retries? I would interpret it that way and did in `YarnFileStageTest#testCopyFromLocalRecursive()`

Reply:
Thanks. Changed to `REMOTE_RESOURCES_FETCH_NUM_RETRY + 1` so that it will do an initial attempt at retry = 0. The retry definition is in line 86, which should only happen when the `FileNotFoundException` happens.

Reply:
And could you clarify by "did in YarnFileStageTest#testCopyFromLocalRecursive()"? Are you suggesting testing the retry on `FileNotFoundException`? I think currently Flink is using a real S3 filesystem instead of a mock, so I am not quite sure how to do it.

Reply:
that was just referring to my way of interpreting what "retry" means - sorry for the confusion